sonar-db/src/main/java/org/sonar/db/permission/PermissionQuery.java

```diff
@@ -127,79 +119,72 @@ public static Builder builder() {
 
   public static class Builder {
     private String permission;
-    private String component;
+    private String componentUuid;
     private String template;
-    private String membership;
-    private String search;
+    private String searchQuery;
+    private boolean withPermissionOnly;
+    private List<String> logins;
+    private List<String> groupNames;
 
     private Integer pageIndex = DEFAULT_PAGE_INDEX;
     private Integer pageSize = DEFAULT_PAGE_SIZE;
 
     private Builder() {
+      // enforce method constructor
     }
 
-    public Builder permission(String permission) {
+    public Builder setPermission(@Nullable String permission) {
+      this.withPermissionOnly = permission != null;
       this.permission = permission;
       return this;
     }
 
-    public Builder template(String template) {
+    public Builder setTemplate(@Nullable String template) {
       this.template = template;
       return this;
     }
 
-    public Builder component(@Nullable String component) {
-      this.component = component;
+    public Builder setComponentUuid(@Nullable String componentUuid) {
+      this.componentUuid = componentUuid;
       return this;
     }
 
-    public Builder membership(@Nullable String membership) {
-      this.membership = membership;
+    public Builder setSearchQuery(@Nullable String s) {
+      this.searchQuery = defaultIfBlank(s, null);
       return this;
     }
 
-    public Builder search(@Nullable String s) {
-      this.search = StringUtils.defaultIfBlank(s, null);
+    public Builder setPageIndex(@Nullable Integer i) {
+      this.pageIndex = i;
       return this;
     }
 
-    public Builder pageSize(@Nullable Integer i) {
+    public Builder setPageSize(@Nullable Integer i) {
       this.pageSize = i;
       return this;
     }
 
-    public Builder pageIndex(@Nullable Integer i) {
-      this.pageIndex = i;
+    public Builder withPermissionOnly() {
+      this.withPermissionOnly = true;
       return this;
     }
 
-    private void initMembership() {
-      if (membership == null) {
-        membership = PermissionQuery.ANY;
-      } else {
-        Preconditions.checkArgument(AVAILABLE_MEMBERSHIP.contains(membership),
-          "Membership is not valid (got " + membership + "). Availables values are " + AVAILABLE_MEMBERSHIP);
-      }
-    }
-
-    private void initPageSize() {
-      if (pageSize == null) {
-        pageSize = DEFAULT_PAGE_SIZE;
-      }
+    public Builder setLogins(@Nullable List<String> logins) {
+      this.logins = logins;
+      return this;
     }
 
-    private void initPageIndex() {
-      if (pageIndex == null) {
-        pageIndex = DEFAULT_PAGE_INDEX;
-      }
-      Preconditions.checkArgument(pageIndex > 0, "Page index must be greater than 0 (got " + pageIndex + ")");
+    public Builder setGroupNames(@Nullable List<String> groupNames) {
+      this.groupNames = groupNames;
+      return this;
     }
 
     public PermissionQuery build() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)